### PR TITLE
Fix the egg variable disaster...

### PR DESCRIPTION
**Pull request by notAreYouScared**
This PR does the following

Updates stock eggs, with new values for the config->file updates
Updates all eggs currently in the database, with migrations
As eggs are imported they're updated, then saved.

If we update them as they're saved, any time the egg is exported it should have the correct variables. 

Closes https://github.com/pelican-dev/panel/issues/317

### CODE REVIEW

**Review by Boy132 — COMMENTED** [review]
https://github.com/pelican-dev/panel/blob/charles/fixing-eggs/app/Services/Eggs/EggConfigurationService.php#L167 needs to be updated as well so `env.VARNAME` is replaced with `server.build.environment.VARNAME` instead of `server.build.env.VARNAME`.

Otherwise looks good.

File: app/Services/Eggs/Sharing/EggUpdateImporterService.php [hunk 1]
```diff
@@ -27,6 +28,18 @@ public function handle(Egg $egg, UploadedFile $file): Egg
...
+            'server.build.env.SERVER_IP' => 'server.allocations.default.ip',
+            'server.build.default.ip' => 'server.allocations.default.ip',
+            'server.build.env.SERVER_PORT' => 'server.allocations.default.port',
+            'server.build.default.port' => 'server.allocations.default.port',
+            'server.build.env.SERVER_MEMORY' => 'server.build.memory_limit',
+            'server.build.memory' => 'server.build.memory_limit',
+            'server.build.env' => 'server.build.environment',
+        ];
+
+        $parsed = $this->recursiveStringReplace($parsed, $replacements);
```
Comment (lancepioch) [inline app/Services/Eggs/Sharing/EggUpdateImporterService.php:41]:
```suggestion
        $parsed = collect($replacements)->reduce(fn ($result, $value, $key) => str_replace($key, $value, $result), $parsed);
```

File: app/Services/Eggs/Sharing/EggUpdateImporterService.php [hunk 2]
```diff
@@ -47,4 +60,16 @@ public function handle(Egg $egg, UploadedFile $file): Egg
...
+        return array_map(function ($value) use ($replacements) {
+            if (is_array($value)) {
+                return $this->recursiveStringReplace($value, $replacements);
+            } elseif (is_string($value)) {
+                return Str::replace(array_keys($replacements), array_values($replacements), $value);
+            }
+
+            return $value;
+        }, $data);
+    }
```
Comment (lancepioch) [inline app/Services/Eggs/Sharing/EggUpdateImporterService.php:74]:
```suggestion
```

File: app/Services/Eggs/Sharing/EggImporterService.php [hunk 3]
```diff
@@ -45,4 +58,17 @@ public function handle(UploadedFile $file): Egg
...
+        return array_map(function ($value) use ($replacements) {
+            if (is_array($value)) {
+                return $this->recursiveStringReplace($value, $replacements);
+            } elseif (is_string($value)) {
+                return Str::replace(array_keys($replacements), array_values($replacements), $value);
+            }
+
+            return $value;
+        }, $data);
+    }
```
Comment (lancepioch) [inline app/Services/Eggs/Sharing/EggImporterService.php:73]:
```suggestion
```

File: app/Services/Eggs/Sharing/EggImporterService.php [hunk 4]
```diff
@@ -25,6 +26,18 @@ public function handle(UploadedFile $file): Egg
...
+            'server.build.env.SERVER_IP' => 'server.allocations.default.ip',
+            'server.build.default.ip' => 'server.allocations.default.ip',
+            'server.build.env.SERVER_PORT' => 'server.allocations.default.port',
+            'server.build.default.port' => 'server.allocations.default.port',
+            'server.build.env.SERVER_MEMORY' => 'server.build.memory_limit',
+            'server.build.memory' => 'server.build.memory_limit',
+            'server.build.env' => 'server.build.environment',
+        ];
+
+        $parsed = $this->recursiveStringReplace($parsed, $replacements);
```
Comment (lancepioch) [inline app/Services/Eggs/Sharing/EggImporterService.php:39]:
```suggestion
        $parsed = collect($replacements)->reduce(fn ($result, $value, $key) => str_replace($key, $value, $result), $parsed);```

**Review by lancepioch — CHANGES_REQUESTED** [review]
I've got a couple small changes that I suggested, also can you please combine the 3 migrations into 1, thanks!

File: app/Services/Eggs/Sharing/EggImporterService.php [hunk 5]
```diff
@@ -25,6 +26,18 @@ public function handle(UploadedFile $file): Egg
     {
         $parsed = $this->parser->handle($file);
 
+        $replacements = [
```
Comment (DjordyKoert) [inline app/Services/Eggs/Sharing/EggImporterService.php:29]: This should be placed inside a `const`, `app/Services/Eggs/Sharing/EggUpdateImporterService.php` uses the same exact same array

File: app/Services/Eggs/Sharing/EggImporterService.php [hunk 6]
```diff
@@ -45,4 +58,17 @@ public function handle(UploadedFile $file): Egg
             return $egg;
         });
     }
+
+    protected function recursiveStringReplace(array $data, array $replacements): array
```
Comment (DjordyKoert) [inline app/Services/Eggs/Sharing/EggImporterService.php:62]: Wouldn't it make sense to place this logic in a trait (or a different class / service)?

Comment (notAreYouScared) [reply app/Services/Eggs/Sharing/EggImporterService.php]: This was removed, and replaced with lances suggestion.

**Review by Boy132 — APPROVED** [review]
LGTM

File: app/Services/Eggs/Sharing/EggImporterService.php (hunk 5 above)
Comment (notAreYouScared) [reply app/Services/Eggs/Sharing/EggImporterService.php:28]: I mean... It works <3

Comment (lancepioch) [reply app/Services/Eggs/Sharing/EggImporterService.php:28]: It should be only in the code once

**Comment by notAreYouScared** [comment]
Looks good, tested this morning.

**Pull request merged**